BoyumFoosballStats/Pages/ScoreCollection/Models/ScoreCollectionViewModel.cs

```diff
@@ -81,7 +82,7 @@ public async Task SaveMatch()
             return;
         }
 
-        match.UpdateMatchesPlayed();
+        match.IncrementMatchesPlayed();
         match.UpdateTrueSkill();
```

Comment:
I think this logic should be moved to another service. I was thinking that we need an abstraction layer so we don't directly call cosmos db. So instead of calling `MatchCrudService` we have an `IMatchHandler` with a `Task<Match> CreateMatch(Match match)` method, in which we can move all the related 'update logic'

Reply:
Or perhaps the `MatchCrudService` could be that 'handler'

Reply:
I'm not sure it belongs in the MatchCrudService unless it directly updates the values in the Cosmos when it is called.
I agree that it's not really an intuitive way to handle this, so maybe something like a handler makes sense.
I think it could be done in a more obvious way at least.

Reply:
@ronnigaba do you plan to do this in this PR? I can also see it as a separate technical issue.

Reply:
> @ronnigaba do you plan to do this in this PR? I can also see it as a separate technical issue.

Let's merge this one and then discuss which option might be the best.